packedForest/src/forestTypes/unsupervisedForests/urerf/unprocessedURerFNode.h

```diff
@@ -0,0 +1,196 @@
+#ifndef rfunprocessedURerFNode_h
+#define rfunprocessedURerFNode_h
+#include "splitURerF.h"
+#include "../baseUnprocessedNodeUnsupervised.h"
+#include "../stratifiedInNodeClassIndicesUnsupervised.h"
+#include <vector>
+#include <random>
+#include <assert.h>
+
+namespace fp{
+
+
+	template <typename T> //
+		class unprocessedURerFNode : public baseUnprocessedNodeUnsupervised<T>{
+			protected:
+
+				splitURerFInfo<T> bestSplitInfo;
+				std::vector< std::vector<int> > featuresToTry;
+
+				//std::random_device rd;
+				//The next three should be static
+				//std::mt19937 rng(rd());   
+				//std::uniform_int_distribution<int> randomMtry(0,fpSingleton::getSingleton().returnMtry()-1);
+				//std::uniform_int_distribution<int> randomFeature(0,fpSingleton::getSingleton().returnNumFeatures()-1);
+				//Example: auto random_integer = uni(rng);
+
+			public:
+				unprocessedURerFNode(int numObsForRoot): baseUnprocessedNodeUnsupervised<T>::baseUnprocessedNodeUnsupervised(numObsForRoot), featuresToTry(fpSingleton::getSingleton().returnMtry()){}
+
+
+				unprocessedURerFNode(int parentID, int dep, bool isLeft): baseUnprocessedNodeUnsupervised<T>::baseUnprocessedNodeUnsupervised(parentID, dep, isLeft), featuresToTry(fpSingleton::getSingleton().returnMtry()){}
+
+
+				~unprocessedURerFNode(){}
+				
+
+
+				inline std::vector<int>& returnBestFeature(){
+					return bestSplitInfo.returnFeatureNum();
+				}
+
+				inline double returnBestImpurity(){
+					return bestSplitInfo.returnImpurity();
+				}
+
+				inline T returnBestCutValue(){
+					return bestSplitInfo.returnSplitValue();
+				}
+
+
+				inline void setBestSplit(splitURerFInfo<T> tempSplit){
+					if(tempSplit.returnImpurity() >= 0){
```

Comment:
reason for this?

Reply:
Agreed.  What is the reason to check that `tempSplit.returnImpurity() >= 0`?